common/djangoapps/third_party_auth/middleware.py

```diff
@@ -23,3 +31,19 @@ def get_redirect_uri(self, request, exception):
             redirect_uri = pipeline.AUTH_DISPATCH_URLS[auth_entry]
 
         return redirect_uri
+
+    def process_exception(self, request, exception):
+        """Handles specific exception raised by Python Social Auth eg HTTPError."""
+
+        referer_url = request.META.get('HTTP_REFERER', '')
+        if referer_url and isinstance(exception, HTTPError) and \
+                exception.response.status_code == 502:
+            referer_url = urlparse.urlparse(referer_url).path
+            if referer_url == '/login':
```

Comment:
Can we use a reverse on this to get the url from the urls  file?

https://docs.djangoproject.com/en/2.0/ref/urlresolvers/